src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -121,20 +153,25 @@ internal RouteEndpointBuilder GetSingleRouteEndpointBuilder()
             return factoryCreatedRequestDelegate(context);
         };
 
-        // The Map methods don't support customizing the order apart from using int.MaxValue to give MapFallback the lowest priority.
-        // Otherwise, we always use the default of 0 unless a convention changes it later.
-        var order = entry.IsFallback ? int.MaxValue : 0;
-
-        RouteEndpointBuilder builder = new(redirectedRequestDelegate, pattern, order)
+        RouteEndpointBuilder builder = new(redirectRequestDelegate, pattern, order)
         {
             DisplayName = displayName,
             ApplicationServices = _applicationServices,
         };
 
-        // Add MethodInfo and HttpMethodMetadata (if any) as first metadata items as they are intrinsic to the route much like
-        // the pattern or default display name. This gives visibility to conventions like WithOpenApi() to intrinsic route details
-        // (namely the MethodInfo) even when applied early as group conventions.
-        builder.Metadata.Add(handler.Method);
+        if (isRouteHandler)
+        {
+            // Add MethodInfo and HttpMethodMetadata (if any) as first metadata items as they are intrinsic to the route much like
+            // the pattern or default display name. This gives visibility to conventions like WithOpenApi() to intrinsic route details
+            // (namely the MethodInfo) even when applied early as group conventions.
+            builder.Metadata.Add(handler.Method);
```

Comment:
I think we want to add this in either case so open api works.

Reply:
I think adding it only for route handlers will actually help with our goal of making `WithOpenApi` something you can only apply on `RouteHandlerBuilder` types even though we removed that constraint. AFAIK, non-route handler scenarios don't need access to the method info.

Reply:
That's the question. Do we want OpenAPI for these endpoints or not? What if someone manually adds `[Consumes]` or `[Produces]` metadata to these endpoints? For now, I'm trying to keep this behavior consistent with the behavior we've already have.

Reply:
I think we should unify these endpoints mostly because of the amount of bugs where people end up calling the RequestDelegate overload and it things stop working.